### PR TITLE
Fix babel config

### PR DESCRIPTION
**Pull request by nickhsine**
### Problem Description
As @taylrj found out, those es5 files (transpiled according to this `babel.config.js`) can not be bundled by webpack.
Even though webpack can build the bundles without any errors and warning, however, the browser will throw error like following one
```
Uncaught ReferenceError: exports is not defined
```

I revert this change for further development.

I will spend time digging it out, and see what is the root cause.

### CODE REVIEW

**Comment by nickhsine** [comment]
Sorry, this is a false alarm.
After retrying, the transipled es modules can be run on browser.

**Comment by nickhsine** [comment]
Need validation.
Close this PR for now.
I will open another one if the problem still existes.

**Comment by nickhsine** [comment]
After digging out with @taylrj , we found that we need to tell webpack to treat files as `commonjs`.
```
{
      '@babel/preset-env',
      {
        useBuiltIns: 'usage',
        corejs: {
          version: 2,
          proposals: true,
        },
        modules: 'commonjs',  // <--- tell webpack to treat all required modules as commonjs
      },
}
```

Otherwise, those files webpack cannot recognize would not have `exports`.
Without `exports`, the codes cannot be run at browser.

As discussion with @taylrj , we both agree that
we can let each pkg import their own `@babel/polyfill` if needed.
Even though the bundle will be bigger, we can use `cacheGroups` to optimize the loading speed.

**Comment by nickhsine** [comment]
@taylrj @YuCJ 
There are two ways to cope with this problem.

## 1. add `modules: 'commonjs'` in root babel.config.js
### Pros: 
1. Only require what we need. We don't need to import the whole `@babel/polyfill`

### Cons:
1. babel.config.js gets harder to understand
2. there are many `require('core-js/')` lines  in the transpiled files,
such as 
```
require("core-js/modules/es6.object.define-property");
require("core-js/modules/es6.function.name");
require("core-js/modules/es6.regexp.to-string");
require("core-js/modules/es6.date.to-string");
require("core-js/modules/es6.object.to-string");
require("core-js/modules/es6.string.iterator");
require("core-js/modules/es6.array.from");
require("core-js/modules/web.dom.iterable");
require("core-js/modules/es7.symbol.async-iterator");
require("core-js/modules/es6.symbol");
require("core-js/modules/es6.array.is-array");
require("core-js/modules/es6.array.map");
```

and those requires actually not necessary if we run those transpiled files in nodejs environment.

## 2. remove `useBuiltIns: 'usage'` in root babel.config.js
In root babel.config.js, we don't care about `@twreporter/polyfill`.
We let each sub-package to worry about this.
In other words, if sub-package needs to support IE, then it imports `@babel/polyfill` either in their `webpack.config.js` or `import '@babel/polyfill'` in the first line of its webpack entry file.

### Pros
1. babel.config.js is easy to understand

### Cons
1. webpack bundle size would be bigger.
2. we need to use `cacheGroups` to optimize the webpack bundles by each sub-pkg.

**Comment by taylrj** [comment]
I vote for option2. If only packages which includes IE as a target need `@babel/polyfill`, they could configure it in their `webpack-config.js`. Therefore, others wouldn't get unused code injected by `babel-loader`. 

There is another concern that option1 may increase the bundle size as [this comment](https://github.com/webpack/webpack/issues/4817#issuecomment-541273101) mentioned. But I haven't found such codesize explosion in my trial though.

**Comment by YuCJ** [comment]
1. No matter we choose 1 or 2, I suggest to use different `@babel/preset-env` setting between building `lib` (without polyfill and `targets: 'node'`) and building `dist` (with polyfill and `targets: /* browserslist query */`). Ex: in [membership-fe](https://github.com/twreporter/membership-fe/blob/44a820e4793c08d4fe4d803ae166cfd1d2194411/webpack.config.js#L66)

2. So I think the main choice is that where we placed the options of `@babel/preset-env` for webpack bundle? In the root `babel.config.js`, or in each package's `babel-loader` option in `webpack.config.js`? I guess if the setting content does not change, the final bundle size will keep the same no matter where we place it? Since the webpack config for each package is independent for now, I'll prefer to place the config in each package as option 2 also.

**Comment by nickhsine** [comment]
> 1. No matter we choose 1 or 2, I suggest to use different `@babel/preset-env` setting between building `lib` (without polyfill and `targets: 'node'`) and building `dist` (with polyfill and `targets: /* browserslist query */`). Ex: in [membership-fe](https://github.com/twreporter/membership-fe/blob/44a820e4793c08d4fe4d803ae166cfd1d2194411/webpack.config.js#L66)

Can we build `lib` by using `targets: 'node'`?

For example,
if we build `@twreporter/scrollable-image` by using `targets:'node'`,
those `lib/` files are transpiled only for node.

Assuming in  `@twreporter/scrollable-image-ui`, it imports `@twreporter/scrollable-image` and tries to bundle `@twreporter/scrollable-images` files by webpack.

In the webpack.config.js, we might write the following config
```
      {
        test: /\.jsx?$/,
        exclude: /node_modules/,
        use: {
          loader: 'babel-loader',
          options: {
            targets: { browsers: [ '>2%', 'ie >= 9' ] },
          },
        },
      },
```

Webpack excludes `node_modules`, in other words, webpack will bundle those transpiled files which are created specifically for nodejs.

If we want to make our npm pkg be isomorphic (both runnable on browser and nodejs), we might need to transpile those files into es modules. (that means, not set `targets: 'node'`)

How do you think?

> 2. So I think the main choice is that where we placed the options of `@babel/preset-env` for webpack bundle? In the root `babel.config.js`, or in each package's `babel-loader` option in `webpack.config.js`? I guess if the setting content does not change, the final bundle size will keep the same no matter where we place it? Since the webpack config for each package is independent for now, I'll prefer to place the config in each package as option 2 also.

**Comment by nickhsine** [comment]
> If we want to make our npm pkg be isomorphic (both runnable on browser and nodejs), we might need to transpile those files into es modules. (that means, not set `targets: 'node'`)

This **es modules** means **es5 scripts**.

Take arrow function for example to clarify my thoughts.

**Arrow functions** supported by nodejs since v4.
However, it's not supported by IE browser (ref: https://caniuse.com/#feat=arrow-functions).

If we have `test.js` like
```javascript

const af = () => {
  console.log('This is a arrow function')
}
```

We transpile `test.js` by `targets: {node: true}` option,
the output will be
```
"use strict";

const af = () => {
  console.log('This is a arrow function');
};
```

If we don't specify `targets`, babel automatically transpile `test.js` to es5;
the output will be 
```
"use strict";

var af = function af() {
  console.log('This is a arrow function');
};
```

In webpack.config.js, we might `exclude` `node_modules` files, which means, we don't use babel-loader to transpile those files under `node_modules`.
Even we have specified which browser (maybe `targets: { browsers: [ '>2%', 'ie >= 9' ] }`) we should support in webpack.config.js, however, the webapck bundles still have some codes (such as arrow functions)  IE might not support.

**Comment by YuCJ** [comment]
@nickhsine 
You're right. I forgot that we will ignore `node_modules` when using webpack `babel-loader`. So if we take option 2, the situation of consumer will be:

- When using `lib` in package: get code without polyfill, the consumer should import polyfill depends on its target environment

- When using `dist` in package: get code which may be with polyfill indisde (depends on the setting of that package)

**Comment by nickhsine** [comment]
> @nickhsine
> You're right. I forgot that we will ignore `node_modules` when using webpack `babel-loader`. So if we take option 2, the situation of consumer will be:
> 
> * When using `lib` in package: get code compiled but without polyfill, the consumer should import polyfill depends on its target environment
> * When using `dist` in package: get code which was compiled and may be with polyfill indisde (depends on the setting of that package)

Yes, if we take option 2, then the consumer's situation would be like you mention here.

If we all agree take option 2, then I am going to merge this PR.

Please add the `:+1` emoji below if you are ok about this change, thx.
